src/components/ChallengeModal/ChallengeModal.tsx

```diff
@@ -907,8 +914,8 @@ export class ChallengeModalBody extends React.Component<
             const tc = updateSystem(
                 this.state.time_control,
                 "simple",
-                this.state.challenge.boardWidth,
-                this.state.challenge.boardHeight,
+                this.state.challenge.boardWidth ?? 19,
```

Comment:
I suspect that these are supposed to have an actual value when the `forceTimeControlSystemIfNecessary` is called, but not 100% sure.